sport/app/football/feed/Competitions.scala

```diff
@@ -136,15 +144,23 @@ trait Competitions extends LiveMatches with Logging with implicits.Collections w
   //one http call updates all competitions
   def refreshCompetitionData() = {
     log.info("Refreshing competition data")
-    FootballClient.competitions.map(_.flatMap { season =>
-      competitionAgents.find(_.competition.id == season.id).map { agent =>
-        val newCompetition = agent.competition.startDate match {
-          case Some(existingStartDate) if season.startDate.isAfter(existingStartDate.toDateTimeAtStartOfDay) => agent.update(agent.competition.copy(startDate = Some(season.startDate)))
-          case None => agent.update(agent.competition.copy(startDate = Some(season.startDate)))
-          case _ =>
+    FootballClient.competitions.map { allComps =>
+      mostRecentCompetitionSeasons(allComps, numToTake = 1).map { season =>
```

Comment:
why not assign a default value to the numToTake param in the function def, so you don't need to specify it here (and anyway it would be 1 most of the time)